docs/developer-docs/latest/developer-resources/database-apis-reference/entity-service/filter.md

```diff
@@ -0,0 +1,357 @@
+
+Results are filtered with the `filters` parameter that accepts [logical operators](#logical-operators) and [attribute operators](#attribute-operators). Every operator should be prefixed with `$`.
+
+## Logical operators
+
+### `$and`
+
+All nested conditions must be true.
+
+**Example**
+
+```js
+const entries = await strapi.entityService.findMany('api::article.article', {
+  filters: {
+    $and: [
+      {
+        title: 'Hello World',
+      },
+      {
+        createdAt: { $gt: '2021-11-17T14:28:25.843Z' },
+      },
+    ],
+  },
+});
+```
+
+`$and` will be used implicitly when passing an object with nested conditions:
+
+```js
+const entries = await strapi.entityService.findMany('api::article.article', {
+  filters: {
+    title: 'Hello World',
+    rating: 12,
+  },
+});
+```
+
+### `$or`
+
+One or many nested conditions must be `true`.
+
+**Example**
+
+```js
+const entries = await strapi.entityService.findMany('api::article.article', {
+  filters: {
+    $or: [
+      {
+        title: 'Hello World',
+      },
+      {
+        title: {
+          $contains: 'Hello',
+        },
+      },
+    ],
+  },
+});
+```
+
+### `$not`
+
+Negates the nested conditions.
+
+**Example**
+
+```js
+const entries = await strapi.entityService.findMany('api::article.article', {
+  filters: {
+    $not: {
+      title: 'Hello World',
+    },
+  },
+});
+```
+
+:::note
+`$not` can be used:
+
+- as a logical operator (e.g. in `filters: { $not: { // conditions… }}`)
+- or [as an attribute operator](#not-2) (e.g. in `filters: { $attribute-name: $not: { … } }`).
+  :::
+
+## Attribute Operators
+
+### `$not`
+
+Negates the nested condition(s).
+
+**Example**
+
+```js
+const entries = await strapi.entityService.findMany('api::article.article', {
+  filters: {
+    title: {
+      $not: {
+        $contains: 'Hello World',
+      },
+    },
+  },
+});
+```
+
+### `$eq`
+
+Attribute equals input value.
+
+**Example**
+
+```js
+const entries = await strapi.entityService.findMany('api::article.article', {
+  filters: {
+    title: {
+      $eq: 'Hello World',
+    },
+  },
+});
+```
+
+`$eq` can be omitted:
+
+```js
+const entries = await strapi.entityService.findMany('api::article.article', {
+  filters: {
+    title: 'Hello World',
+  },
+});
+```
+
+### `$ne`
+
+Attribute does not equal input value.
+
+**Example**
+
+```js
+const entries = await strapi.entityService.findMany('api::article.article', {
+  filters: {
+    title: {
+      $ne: 'ABCD',
+    },
+  },
+});
+```
+
+### `$in`
+
+Attribute is contained in the input list.
+
+**Example**
+
+```js
+const entries = await strapi.entityService.findMany('api::article.article', {
+  filters: {
+    title: {
+      $in: ['Hello', 'Hola', 'Bonjour'],
+    },
+  },
+});
+```
+
+`$in` can be ommited when passing an array of values:
+
+```js
+const entries = await strapi.entityService.findMany('api::article.article', {
+  filters: {
+    title: ['Hello', 'Hola', 'Bonjour'],
+  },
+});
+```
+
+### `$nin`
```

Comment:
it's actually `notIn`

Here is the list of valid operators 

 `$not`
`$in`
`$notIn`
`$eq`
`$ne`
`$gt`
`$gte`
`$lt`
`$lte`
`$null`
`$notNull`
`$between`
`$startsWith`
`$endsWith`
`$contains`
`$notContains`
`$containsi`
`$notContainsi`

Reply:
Thanks @alexandrebodin 
`$containsi` is **not** case-sensitive (`i` for insensitive? 🤔) while `$contains` is case-sensitive, or is it the other way round?

Reply:
And I guess [this list of filters for REST API](https://docs-next.strapi.io/developer-docs/latest/developer-resources/database-apis-reference/rest-api.html#filters) also needs to be updated (ncontains → notContains, etc.)?

Reply:
Yes you are right on all points